core/chains/evm/chain.go

```diff
@@ -142,22 +142,39 @@ type AppConfig interface {
 
 type ChainRelayExtenderConfig struct {
 	Logger   logger.Logger
-	DB       *sqlx.DB
 	KeyStore keystore.Eth
-	*RelayerConfig
+	*ChainOpts
 }
 
-// options for the relayer factory.
-// TODO BCF-2508 clean up configuration of chain and relayer after BCF-2440
-// the factory wants to own the logger and db
-// the factory creates extenders, which need the same and more opts
-type RelayerConfig struct {
+func (c ChainRelayExtenderConfig) Validate() error {
+	var err error
+	if c.Logger == nil {
+		err = errors.Join(err, fmt.Errorf("nil Logger"))
+	}
+	if c.KeyStore == nil {
+		err = errors.Join(err, fmt.Errorf("nil Keystore"))
+	}
+	if c.ChainOpts == nil {
+		err = errors.Join(err, fmt.Errorf("nil ChainOpts"))
+	} else {
+		err = errors.Join(err, c.ChainOpts.Validate())
```

Comment:
If you use a pointer receiver, then you could bundle this nil check inside of the method:
```go
func (o *ChainOpts) Validate() error {
	if o == nil {
		return errors.New("nil ChainOpts")
	}
}
```